SDK_versioned_docs/version-3.0.0/guides/04-fetching-prices.md

```diff
@@ -0,0 +1,146 @@
+---
+id: fetching-prices
+title: Fetching Spot Prices
+---
+
+## Fetching Token Prices with the SDK
+
+To calculate the spot price of any pool, you can call trivially call `token0Price` or `token1Price` on any Pool object (how nice!). However, in addition to learning a couple of simple steps needed to fetch token prices, you will also learn *how* the SDK calculates these quantities. Going through these calculations will hopefully provide the necessary context behind fixed point numbers, square roots, and difficult-to-understand variable names like `sqrtPriceX96`. :)
+
+### Calling the functions
+
+Similar to other examples, you first must set up your pool. If you’re unsure how to collect all the parameters necessary in creating a Pool instance see [Creating a Pool Instance](https://docs.uniswap.org/sdk/guides/creating-a-pool) or look at this typescript example [here](https://github.com/Uniswap/uniswap-docs/blob/main/sdk-examples/AddAndRemoveLiquidity.tsx). The `Pool` class contains two getter methods `token0Price` and `token1Price` which will return the prices of each token respectively as a `Price`.
+
+After constructing the pool, you can save the token prices as constants:
+
+```typescript
+  const DAI_USDC_POOL = new Pool(
+    DAI,
+    USDC,
+    immutables.fee,
+    state.sqrtPriceX96.toString(),
+    state.liquidity.toString(),
+    state.tick
+  )
+  
+  const token0Price =  DAI_USDC_POOL.token0Price
+  const token1Price = DAI_USDC_POOL.token1Price
+```
+
+### Understanding sqrtPrice
+
+What is `sqrtPriceX96`?
+
+In Uniswap V3, prices of tokens are stored (rather than derived) to allow pools to perform higher precision operations. In the actual implementation, prices are stored as square roots, hence the variable name `sqrtPriceX96`. The price is stored as a square root because of the geometric nature of the core AMM algorithm, x*y=k. Essentially, the math works out well when working with the square root of the price. 
```

Comment:
maybe change to "...hence the `sqrt` prefix."

could also add a link to the whitepaper as a reference for the math working out well with square roots.